### DISCUSSION TITLE
Big problem for using trustwallet for desktop project !!

### DISCUSSION THREAD

**Discussion by guardesign** [trustwallet/wallet-core#1594] [Q&A]
**hi everybody
we want to use trust wallet core for developing wallet in our project but we encounter some issues  : 
we had able to build a native core wallet in Linux and our problem is when we want to use (Sample Go Integration for Wallet-Core) native besides that for using repositories we faced up error !!!**

//////////////////////////////////////////////////////////////////////

 ./include/TrustWalletCore/TWBase.h:47:18: error: missing binary operator before token "("
   47 | #if __has_feature(assume_nonnull)
      |                  ^
./include/TrustWalletCore/TWBase.h:56:19: error: missing binary operator before token "("
   56 | #if !__has_feature(nullability)
      |                   ^
In file included from ./include/TrustWalletCore/TWPublicKeyType.h:9,
                 from ./include/TrustWalletCore/TWPublicKey.h:11,
                 from ./include/TrustWalletCore/TWPrivateKey.h:12,
                 from ./include/TrustWalletCore/TWCoinType.h:14,
                 from ./include/TrustWalletCore/TWHDWallet.h:10,
                 from ./main.go:5:
./include/TrustWalletCore/TWBase.h:47:18: error: missing binary operator before token "("
   47 | #if __has_feature(assume_nonnull)
      |                  ^
./include/TrustWalletCore/TWBase.h:56:19: error: missing binary operator before token "("
   56 | #if !__has_feature(nullability)
      |                   ^
In file included from ./include/TrustWalletCore/TWString.h:9,
                 from ./include/TrustWalletCore/TWPublicKey.h:12,
                 from ./include/TrustWalletCore/TWPrivateKey.h:12,
                 from ./include/TrustWalletCore/TWCoinType.h:14,
                 from ./include/TrustWalletCore/TWHDWallet.h:10,
                 from ./main.go:5:
./include/TrustWalletCore/TWBase.h:47:18: error: missing binary operator before token "("
   47 | #if __has_feature(assume_nonnull)
      |                  ^
./include/TrustWalletCore/TWBase.h:56:19: error: missing binary operator before token "("
   56 | #if !__has_feature(nullability)
      |                   ^
In file included from ./include/TrustWalletCore/TWPurpose.h:9,
                 from ./include/TrustWalletCore/TWCoinType.h:15,
                 from ./include/TrustWalletCore/TWHDWallet.h:10,
                 from ./main.go:5:
./include/TrustWalletCore/TWBase.h:47:18: error: missing binary operator before token "("
   47 | #if __has_feature(assume_nonnull)
      |                  ^
./include/TrustWalletCore/TWBase.h:56:19: error: missing binary operator before token "("
   56 | #if !__has_feature(nullability)
      |                   ^
 ./include/TrustWalletCore/TWBase.h:56:19: error: missing binary operator before token "("

///////////////////////////////////////////////////////////////////

is there any solution to this problem !!

Best

**Comment by 0x00trust** [comment]
Recommend used mac Os and this IDE tool 
Appcode
https://www.jetbrains.com/objc/download/
Build 1 hour for lib only

**Reply by guardesign** [reply]
Hi
Thank you for your answer, but I want to use trust wallet for creating a wallet for my project (cryptocurrency exchange) for desktop, my hosting, and server-based on Linux operating system, could you help me !!

**Reply by guardesign** [reply]
I want to create a wallet in Linux native and use Go programming to use it inside my project and I don't wanna use docker inside our project. after we build native Linux and use GO for deploying we encounter an error I sent it before ... could you please help me with that !?

**Reply by 0x00trust** [reply]
Sorry I cannot 
Trust wallet is part of binance.
And inside connect binance, to crate multiple coin address  but never give for you private key.
You got only seed phara of ETH only 
Its True

**Reply by guardesign** [reply]
hi
Thank you for your answer
could you please give me a recommendation about our project !! we want to develop a secure wallet for our exchange ! is there any secure library or something like a trust wallet for developing you can recommend ? just for helping us

**Reply by 0x00trust** [reply]
> hi
> Thank you for your answer
> could you please give me a recommendation about our project !! we want to develop a secure wallet for our exchange ! is there any secure library or something like a trust wallet for developing you can recommend ? just for helping us

You can use metamask 
https://github.com/MetaMask/metamask-mobile
100% opensource 
You can use code logo and anthing you need
